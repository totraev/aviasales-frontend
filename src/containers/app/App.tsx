import React, { Component } from 'react';
import styles from './App.module.css';

import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import { compose } from 'recompose';

import Header from '../../components/header/Header';
import Filter from '../filter/Filter';
import TicketList from '../ticketList/TicketList';

import { fetchTickets } from '../../redux/modules/tickets';
import { IDispatchProps, IProps } from './App.types';

class App extends Component<IProps> {
  public componentDidMount() {
    this.props.fetchTickets();
  }

  public render() {
    return (
      <div className={styles.container}>
        <Header />

        <div className={styles.contentWrap}>
          <div className={styles.sidebar}>
            <Filter />
          </div>

          <div className={styles.content}>
            <TicketList />
          </div>
        </div>
      </div>
    );
  }
}

export default compose<IDispatchProps, {}>(
  hot(module),
  connect<{}, IDispatchProps>(null, { fetchTickets }),
)(App);
