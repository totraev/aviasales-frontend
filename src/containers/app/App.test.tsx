import React from 'react';
import renderer from 'react-test-renderer';

import { App } from './App';

describe('App component', () => {
  it('Should renders correctly', () => {
    const fetchTickets = () => null;
    const tree = renderer.create(<App fetchTickets={fetchTickets}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
