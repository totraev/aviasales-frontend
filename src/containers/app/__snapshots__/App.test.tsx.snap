// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <img
      alt="Aviasales"
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "logo.svg",
        }
      }
    />
  </div>
  <div
    className="contentWrap"
  >
    <div
      className="sidebar"
    >
      <div
        className="filter"
      >
        <h3
          className="title"
        >
          ВАЛЮТА
        </h3>
        <div
          className="toggle"
        >
          <button
            className="btnDefault"
            onClick={[Function]}
          >
            RUB
          </button>
          <button
            className="btnActive"
            onClick={[Function]}
          >
            USD
          </button>
          <button
            className="btnDefault"
            onClick={[Function]}
          >
            EUR
          </button>
        </div>
        <h3
          className="title"
        >
          КОЛИЧЕСТВО ПЕРЕСАДОК
        </h3>
        <div
          className="stops"
        >
          <div
            className="wrap"
          >
            <label
              className="label"
              htmlFor="stops_0"
            >
              <input
                checked={true}
                className="checkbox"
                id="stops_0"
                onClick={[Function]}
                type="checkbox"
              />
              <span
                className="faceChecked"
              />
              БЕЗ ПЕРЕСАДОК
            </label>
            <div
              className="append"
            >
              <button
                className="uncheckOther"
                onClick={[Function]}
              >
                ТОЛЬКО
              </button>
            </div>
          </div>
          <div
            className="wrap"
          >
            <label
              className="label"
              htmlFor="stops_1"
            >
              <input
                checked={true}
                className="checkbox"
                id="stops_1"
                onClick={[Function]}
                type="checkbox"
              />
              <span
                className="faceChecked"
              />
              1 ПЕРЕСАДКА
            </label>
            <div
              className="append"
            >
              <button
                className="uncheckOther"
                onClick={[Function]}
              >
                ТОЛЬКО
              </button>
            </div>
          </div>
          <div
            className="wrap"
          >
            <label
              className="label"
              htmlFor="stops_2"
            >
              <input
                checked={true}
                className="checkbox"
                id="stops_2"
                onClick={[Function]}
                type="checkbox"
              />
              <span
                className="faceChecked"
              />
              2 ПЕРЕСАДКИ
            </label>
            <div
              className="append"
            >
              <button
                className="uncheckOther"
                onClick={[Function]}
              >
                ТОЛЬКО
              </button>
            </div>
          </div>
          <div
            className="wrap"
          >
            <label
              className="label"
              htmlFor="stops_3"
            >
              <input
                checked={true}
                className="checkbox"
                id="stops_3"
                onClick={[Function]}
                type="checkbox"
              />
              <span
                className="faceChecked"
              />
              3 ПЕРЕСАДКИ
            </label>
            <div
              className="append"
            >
              <button
                className="uncheckOther"
                onClick={[Function]}
              >
                ТОЛЬКО
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="content"
    >
      <div
        className="list"
      >
        <div
          className="ticket"
        >
          <div
            className="carrier"
          >
            <img
              alt="TK"
              className="logo"
              src="logo@3x.png"
              srcSet="
          logo.png,
          logo@2x.png 2x,
          logo@3x.png
        "
            />
            <button
              className="btn"
            >
              Купить
              <br />
              за 
              21300
               ​
              ₽
            </button>
          </div>
          <div
            className="info"
          >
            <div
              className="segment"
            >
              <span
                className="time"
              >
                17:20
              </span>
              <div
                className="seporator"
              >
                <div
                  className="stops"
                >
                  1 ПЕРЕСАДКА
                </div>
                <div
                  className="hr"
                >
                  <div
                    className="line"
                  />
                  <span
                    className="airplane"
                  />
                </div>
              </div>
              <span
                className="time"
              >
                23:50
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="origin"
              >
                VVO
                , 
                Владивосток
              </span>
              <span
                className="destination"
              >
                Тель-Авив
                , 
                TLV
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="date"
              >
                12.05.18
              </span>
              <span
                className="date"
              >
                12.05.18
              </span>
            </div>
          </div>
        </div>
        <div
          className="ticket"
        >
          <div
            className="carrier"
          >
            <img
              alt="TK"
              className="logo"
              src="logo@3x.png"
              srcSet="
          logo.png,
          logo@2x.png 2x,
          logo@3x.png
        "
            />
            <button
              className="btn"
            >
              Купить
              <br />
              за 
              21300
               ​
              ₽
            </button>
          </div>
          <div
            className="info"
          >
            <div
              className="segment"
            >
              <span
                className="time"
              >
                17:20
              </span>
              <div
                className="seporator"
              >
                <div
                  className="stops"
                >
                  1 ПЕРЕСАДКА
                </div>
                <div
                  className="hr"
                >
                  <div
                    className="line"
                  />
                  <span
                    className="airplane"
                  />
                </div>
              </div>
              <span
                className="time"
              >
                23:50
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="origin"
              >
                VVO
                , 
                Владивосток
              </span>
              <span
                className="destination"
              >
                Тель-Авив
                , 
                TLV
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="date"
              >
                12.05.18
              </span>
              <span
                className="date"
              >
                12.05.18
              </span>
            </div>
          </div>
        </div>
        <div
          className="ticket"
        >
          <div
            className="carrier"
          >
            <img
              alt="TK"
              className="logo"
              src="logo@3x.png"
              srcSet="
          logo.png,
          logo@2x.png 2x,
          logo@3x.png
        "
            />
            <button
              className="btn"
            >
              Купить
              <br />
              за 
              21300
               ​
              ₽
            </button>
          </div>
          <div
            className="info"
          >
            <div
              className="segment"
            >
              <span
                className="time"
              >
                17:20
              </span>
              <div
                className="seporator"
              >
                <div
                  className="stops"
                >
                  1 ПЕРЕСАДКА
                </div>
                <div
                  className="hr"
                >
                  <div
                    className="line"
                  />
                  <span
                    className="airplane"
                  />
                </div>
              </div>
              <span
                className="time"
              >
                23:50
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="origin"
              >
                VVO
                , 
                Владивосток
              </span>
              <span
                className="destination"
              >
                Тель-Авив
                , 
                TLV
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="date"
              >
                12.05.18
              </span>
              <span
                className="date"
              >
                12.05.18
              </span>
            </div>
          </div>
        </div>
        <div
          className="ticket"
        >
          <div
            className="carrier"
          >
            <img
              alt="TK"
              className="logo"
              src="logo@3x.png"
              srcSet="
          logo.png,
          logo@2x.png 2x,
          logo@3x.png
        "
            />
            <button
              className="btn"
            >
              Купить
              <br />
              за 
              21300
               ​
              ₽
            </button>
          </div>
          <div
            className="info"
          >
            <div
              className="segment"
            >
              <span
                className="time"
              >
                17:20
              </span>
              <div
                className="seporator"
              >
                <div
                  className="stops"
                >
                  1 ПЕРЕСАДКА
                </div>
                <div
                  className="hr"
                >
                  <div
                    className="line"
                  />
                  <span
                    className="airplane"
                  />
                </div>
              </div>
              <span
                className="time"
              >
                23:50
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="origin"
              >
                VVO
                , 
                Владивосток
              </span>
              <span
                className="destination"
              >
                Тель-Авив
                , 
                TLV
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="date"
              >
                12.05.18
              </span>
              <span
                className="date"
              >
                12.05.18
              </span>
            </div>
          </div>
        </div>
        <div
          className="ticket"
        >
          <div
            className="carrier"
          >
            <img
              alt="TK"
              className="logo"
              src="logo@3x.png"
              srcSet="
          logo.png,
          logo@2x.png 2x,
          logo@3x.png
        "
            />
            <button
              className="btn"
            >
              Купить
              <br />
              за 
              21300
               ​
              ₽
            </button>
          </div>
          <div
            className="info"
          >
            <div
              className="segment"
            >
              <span
                className="time"
              >
                17:20
              </span>
              <div
                className="seporator"
              >
                <div
                  className="stops"
                >
                  1 ПЕРЕСАДКА
                </div>
                <div
                  className="hr"
                >
                  <div
                    className="line"
                  />
                  <span
                    className="airplane"
                  />
                </div>
              </div>
              <span
                className="time"
              >
                23:50
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="origin"
              >
                VVO
                , 
                Владивосток
              </span>
              <span
                className="destination"
              >
                Тель-Авив
                , 
                TLV
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="date"
              >
                12.05.18
              </span>
              <span
                className="date"
              >
                12.05.18
              </span>
            </div>
          </div>
        </div>
        <div
          className="ticket"
        >
          <div
            className="carrier"
          >
            <img
              alt="TK"
              className="logo"
              src="logo@3x.png"
              srcSet="
          logo.png,
          logo@2x.png 2x,
          logo@3x.png
        "
            />
            <button
              className="btn"
            >
              Купить
              <br />
              за 
              21300
               ​
              ₽
            </button>
          </div>
          <div
            className="info"
          >
            <div
              className="segment"
            >
              <span
                className="time"
              >
                17:20
              </span>
              <div
                className="seporator"
              >
                <div
                  className="stops"
                >
                  1 ПЕРЕСАДКА
                </div>
                <div
                  className="hr"
                >
                  <div
                    className="line"
                  />
                  <span
                    className="airplane"
                  />
                </div>
              </div>
              <span
                className="time"
              >
                23:50
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="origin"
              >
                VVO
                , 
                Владивосток
              </span>
              <span
                className="destination"
              >
                Тель-Авив
                , 
                TLV
              </span>
            </div>
            <div
              className="segment"
            >
              <span
                className="date"
              >
                12.05.18
              </span>
              <span
                className="date"
              >
                12.05.18
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
